bokeh/models/text.py

```diff
@@ -69,6 +70,11 @@ class MathText(BaseText):
     Base class for renderers of mathematical content.
     """
 
+    @staticmethod
+    def is_math_text_string(text: Unknown) -> bool:
+        return isinstance(text, str) and text.startswith("$") and text.endswith("$")
```

Comment:
Isn't this awkward? Properties themselves are supposed to for for validation of types and now we are having to bold static method predicates like this in places.

Reply:
I would argue that mathstrings are not much of a Property, they are more like a string interpretation that need a parser of some kind

Reply:
Ideally we just always send ascii strings and let BokehJS sort it out. But unless we are just going to always include the mathjax bundle [1], we are bound by the constraint that we need to be able to detect the presence of mathtext on the python side to condition the inclusion of the mathjax bundle.

> I would argue that mathstrings are not much of a Property, they are more like a string interpretation that need a parser of some kind

How is this differnet from the `Regex` property that validates according to a regex match? We have a sophisticated tpye system that allows for applying validations and transforms in just the correct places. We should utilize that instead of fighting it.

Reply:
`Regex` don't transform/convert themselves.

But just that difference does not justify the use of an alternative type system like I did here.

I will implement the `converts` methods in type system that looks to me the cleanest solution.